lib/cog_api/http/roles.ex

```diff
@@ -7,8 +7,8 @@ defmodule CogApi.HTTP.Roles do
     Base.get(endpoint, "roles") |> Base.format_response("roles", [%Role{}])
   end
 
-  def show(%Endpoint{}=endpoint, id) do
-    Base.get(endpoint, "roles/#{id}") |> Base.format_response("role", %Role{})
+  def show(%Endpoint{}=endpoint, role_name) do
+    Base.get_by(endpoint, "roles", name: role_name) |> Base.format_response("role", %Role{})
   end
```

Comment:
For Flywheel we will have the id at this point, and would prefer to use that. Obviously that's a bit different for `cogctl`, which will probably still need to pass the name. Can we allow this to accept either (perhaps as a hash) and do the right thing accordingly?

It's likely this will have to be done for all `show`/`delete`/`update` functions.